pkgs/unified_analytics/lib/src/event.dart

```diff
@@ -716,11 +724,67 @@ final class Event {
       other is Event &&
       other.runtimeType == runtimeType &&
       other.eventName == eventName &&
-      compareEventData(other.eventData, eventData);
-
-  @override
-  String toString() => jsonEncode({
+      _deepCollectionEquality.equals(other.eventData, eventData);
+
+  /// Converts an instance of [Event] to JSON.
+  ///
+  /// Example for [Event.timing] converted to JSON below.
+  /// ```json
+  /// {
+  ///   "eventName": "timing",
+  ///   "eventData": {
+  ///     "workflow": "my-work-flow",
+  ///     "variableName": "my-variable",
+  ///     "elapsedMilliseconds": 123,
+  ///     "label": "my-label"
+  ///   }
+  /// }
+  /// ```
+  String toJson() => jsonEncode({
         'eventName': eventName.label,
         'eventData': eventData,
       });
+
+  @override
+  String toString() => toJson();
+
+  /// Returns a valid instance of [Event] if [json] follows the correct schema.
+  ///
+  /// Schema is below with the following notes:
+  /// - The `eventName` key must have a string value that exists in [DashEvent].
+  /// - The `eventData` key must have a nested object as its value.
+  /// ```json
+  /// {
+  ///   "eventName": "string-label",
+  ///   "eventData": {
+  ///     "myVar1": "value1",
+  ///     "myVar2": 123
+  ///   }
+  /// }
+  /// ```
```

Comment:
I think we can remove this part of the dartdoc, as it would be difficult to ensure this stays up to date if the schema changes, and the source of truth should be the code.

Instead, I think it would be helpful to future readers, to give some context as to the use case of deserializing an event (my understanding is that the primary use case today is that devtools server needs to receive from the devtools app all events, check that the user is opted in, and then actually make the HTTP request to the google analytics server).